File: src/horizon_mcp/auth/apikey.py
"""API Key authentication provider (X-API-ID / X-API-KEY headers)."""

from __future__ import annotations

from horizon_mcp.auth.base import AuthProvider


class ApiKeyAuthProvider(AuthProvider):
    """Authenticate via static API ID and API Key headers."""

    def __init__(self, api_id: str, api_key: str) -> None:
        if not api_id or not api_key:
            raise ValueError(
                "HORIZON_API_ID and HORIZON_API_KEY must be set. "
                "See .env.example for configuration."
            )
        self._api_id = api_id
        self._api_key = api_key

    async def get_headers(self) -> dict[str, str]:
        return {"X-API-ID": self._api_id, "X-API-KEY": self._api_key}

    async def refresh_if_needed(self) -> None:
        pass  # Static credentials  -  nothing to refresh

File: src/auth/apikey.ts
import { AuthProvider } from './base.js';

/**
 * Authenticate via static API ID and API Key headers.
 * Injects X-API-ID and X-API-KEY custom Horizon headers.
 */
export class ApiKeyAuthProvider extends AuthProvider {
  private readonly _apiId: string;
  private readonly _apiKey: string;

  constructor(apiId: string, apiKey: string) {
    super();
    if (!apiId || !apiKey) {
      throw new Error(
        'HORIZON_API_ID and HORIZON_API_KEY must both be set. ' +
          'See .env.example for configuration.',
      );
    }
    this._apiId = apiId;
    this._apiKey = apiKey;
  }

  async getHeaders(): Promise<Record<string, string>> {
    return { 'X-API-ID': this._apiId, 'X-API-KEY': this._apiKey };
  }

  async refreshIfNeeded(): Promise<void> {
    // Static credentials - nothing to refresh
  }
}
